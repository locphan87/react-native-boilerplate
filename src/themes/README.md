# Themes

Application specific themes
* Base Styles
* Fonts
* Metrics
* Colors

