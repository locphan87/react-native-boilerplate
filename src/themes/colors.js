export default {
  base: 'rgba(31, 32, 39, 1)',
  white: 'white',
  yellow: 'yellow',
  gray: '#959299'
}
