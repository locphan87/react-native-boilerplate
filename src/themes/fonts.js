const SIZE = {
  h1: 20,
  h2: 18,
  h3: 16,
  _: 14
}

export default {
  SIZE
}
