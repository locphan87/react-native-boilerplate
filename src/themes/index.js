export { default as COLORS } from './colors'
export { default as FONTS } from './fonts'
export { default as MAIN_THEME } from './theme'
