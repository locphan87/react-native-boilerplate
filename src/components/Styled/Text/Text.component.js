// @flow
import { Text } from 'react-native'
import glamorous from 'glamorous-native'
import { keys, find } from 'ramda'

import { match } from '../../../utils/common.util'
import { FONTS } from '../../../themes'

const styles = (props, { colors }) => {
  const hasStyleInProps = key => props[key]
  const fontStyle = find(hasStyleInProps, keys(FONTS.SIZE))
  const fontSize = match(fontStyle)(FONTS.SIZE)

  return {
    color: props.color || colors.white,
    fontSize
  }
}

const StyledText = glamorous(Text)(styles)
StyledText.displayName = 'StyledText'

export default StyledText
