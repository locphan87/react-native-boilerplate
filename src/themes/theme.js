import colors from './colors'

const mainTheme = {
  colors
}

export default mainTheme
