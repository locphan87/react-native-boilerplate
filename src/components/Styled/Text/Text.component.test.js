import React from 'react'
import { shallow } from 'enzyme'

import { snapTest } from '../../../utils/test.util'
import { MAIN_THEME } from '../../../themes'

import Text from './Text.component'

const props = {
  h1: true,
  theme: MAIN_THEME
}
const wrapper = shallow(<Text {...props} />)

describe('Styled Text', () => {
  snapTest(wrapper, [
    {
      props,
      description: 'basic render'
    }
  ])
})
