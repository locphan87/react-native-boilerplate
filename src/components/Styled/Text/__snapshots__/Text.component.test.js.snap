// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Text snapshots basic render 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  h1={true}
  style={
    Array [
      Object {
        "color": "white",
        "fontSize": 20,
      },
    ]
  }
/>
`;
